client/src/components/App/App.js

```diff
@@ -21,19 +21,47 @@ class App extends Component {
       newUsername: "",
       newEmail: "",
       newPassword: "",
-      newPasswordConfirm: ""
-
+      newPasswordConfirm: "",
+      signupError: "",
+      disabledProp: true
     };
     this.handleChange = this.handleChange.bind(this);
     this.handleLogin = this.handleLogin.bind(this);
     this.handleRegister = this.handleRegister.bind(this);
   }
 
   handleChange(event) {
-
     const { name } = event.target;
     const { value } = event.target;
     this.setState({ [name]: value });
+    let errorMessage = "";
+    let disabled = true;
+    if (
+      this.state.newUsername === "" &&
```

Comment:
also use `.trim()` on the strings so users can't input empty spaces as a username and `.trim()` just removes these empty spaces, and if the user only inputed empty spaces then `trim()` is going to trim it down to empty string